runtime/src/main/cpp/Memory.cpp

```diff
@@ -577,7 +578,7 @@ inline void DecrementRC(ContainerHeader* container, bool useCycleCollector) {
         container->setBuffered();
         auto state = memoryState;
         state->toFree->push_back(container);
-        if (state->gcSuspendCount == 0 && freeableSize(state) >= state->gcThreshold) {
+        if (state->gcSuspendCount == 0 && !state->finalizationFromGCInProgress && freeableSize(state) >= state->gcThreshold) {
```

Comment:
Why not reuse gcSuspendCount for this purpose?

Reply:
Good point.